### PR TITLE
Eagerly resolve toolchains to canonical, fixed reference

### PR DESCRIPTION
**Pull request by Kha**
This also gets rid of the `update-hashes/` directory (it won't literally get rid of it on existing installs but the contents do not matter anymore).

### CODE REVIEW

File: src/elan-cli/elan_mode.rs [hunk 1]
```diff
@@ -344,13 +320,8 @@ fn show(cfg: &Cfg) -> Result<()> {
...
             print_header("installed toolchains")
         }
-        let default_name = cfg.get_default()?;
         for t in installed_toolchains {
-            if default_name.as_ref() == Some(&t) {
-                println!("{} (default)", t);
-            } else {
-                println!("{}", t);
-            }
+            println!("{}", t);
```
Comment (digama0) [inline src/elan-cli/elan_mode.rs:324]: Why doesn't it show the default anymore? This was a useful functionality and one of the reasons I use `elan show`.

Comment (Kha) [reply src/elan-cli/elan_mode.rs:324]: To me the main use of `show` is to document the choice of active toolchain, where `default` is still displayed. Do you have other use cases in mind? Should `elan default` without argument show the (unresolved) option value?

**Comment by tydeu** [comment]
I would love to see this change! Avoiding multiple installs of the same toolchain under different names would be wonderful (and save me disk space). What will be procedure for users updating elan while already having such toolchain installed (and ones for stable/nightly)?

**Comment by Kha** [comment]
> What will be procedure for users updating elan while already having such toolchain installed (and ones for stable/nightly)?

I would prefer the same simple strategy as getting rid of unused toolchains so far: `rm` :)

File: src/elan-cli/elan_mode.rs (hunk 1 above)
Comment (Kha) [reply src/elan-cli/elan_mode.rs:323]: It shows the default again

**Pull request merged**